components/brave_wallet/browser/swap_response_parser.cc

```diff
@@ -60,77 +123,455 @@ bool ParseSwapResponse(const std::string& json,
+  if (!minimum_protocol_fee)
+    return nullptr;
+
+  swap_response->minimum_protocol_fee = *minimum_protocol_fee;
+
+  auto buy_token_address =
+      ParseResultFromDict(response_dict, "buyTokenAddress");
+  if (!buy_token_address)
+    return nullptr;
+  swap_response->buy_token_address = *buy_token_address;
+
+  auto sell_token_address =
+      ParseResultFromDict(response_dict, "sellTokenAddress");
+  if (!sell_token_address)
+    return nullptr;
+  swap_response->sell_token_address = *sell_token_address;
+
+  auto buy_amount = ParseResultFromDict(response_dict, "buyAmount");
+  if (!buy_amount)
+    return nullptr;
+  swap_response->buy_amount = *buy_amount;
+
+  auto sell_amount = ParseResultFromDict(response_dict, "sellAmount");
+  if (!sell_amount)
+    return nullptr;
+  swap_response->sell_amount = *sell_amount;
+
+  auto allowance_target = ParseResultFromDict(response_dict, "allowanceTarget");
+  if (!allowance_target)
+    return nullptr;
+  swap_response->allowance_target = *allowance_target;
+
+  auto sell_token_to_eth_rate =
+      ParseResultFromDict(response_dict, "sellTokenToEthRate");
+  if (!sell_token_to_eth_rate)
+    return nullptr;
+  swap_response->sell_token_to_eth_rate = *sell_token_to_eth_rate;
+
+  auto buy_token_to_eth_rate =
+      ParseResultFromDict(response_dict, "buyTokenToEthRate");
+  if (!buy_token_to_eth_rate)
+    return nullptr;
+  swap_response->buy_token_to_eth_rate = *buy_token_to_eth_rate;
+
+  return swap_response;
+}
+
+mojom::JupiterSwapQuotePtr ParseJupiterSwapQuote(const std::string& json) {
+  //  {
+  //    "data": [
+  //      {
+  //        "inAmount": 10000,
+  //        "outAmount": 261273,
+  //        "amount": 10000,
+  //        "otherAmountThreshold": 258660,
+  //        "outAmountWithSlippage": 258660,
+  //        "swapMode": "ExactIn",
+  //        "priceImpactPct": 0.008955716118219659,
+  //        "marketInfos": [
+  //          {
+  //            "id": "2yNwARmTmc3NzYMETCZQjAE5GGCPgviH6hiBsxaeikTK",
+  //            "label": "Orca",
+  //            "inputMint": "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
+  //            "outputMint": "MNDEFzGvMt87ueuHvVU9VcTqsAP5b3fTGPsHuuPA5ey",
+  //            "notEnoughLiquidity": false,
+  //            "inAmount": 10000,
+  //            "outAmount": 117001203,
+  //            "priceImpactPct": 1.196568750220778e-7,
+  //            "lpFee": {
+  //              "amount": 30,
+  //              "mint": "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
+  //              "pct": 0.003
+  //            },
+  //            "platformFee": {
+  //              "amount": 0,
+  //              "mint": "MNDEFzGvMt87ueuHvVU9VcTqsAP5b3fTGPsHuuPA5ey",
+  //              "pct": 0
+  //            }
+  //          }
+  //        ]
+  //      }
+  //    ],
+  //    "timeTaken": 0.044471802000089156
+  //  }
+
+  // STEP 1: convert JSON integer fields in routes to string values.
+  auto converted_json = ConvertMultiUint64InObjectArrayToString(
```

Comment:
Converting here is wrong, for responses, you need to bind conversion function when calling api_request_helper_ to do the conversion before sanitization.
This will be caught if you have swap service unit test for this API.
Unit test of ParseJupiterSwapQuote won't catch this because it doesn't include the sanitization step.

Reply:
See JsonRpcService::GetSolanaSignatureStatuses and solana::ParseGetSignatureStatuses for reference, and JsonRpcServiceUnitTest::GetSolanaSignatureStatuses for test.

Reply:
Thanks for the references. So in this particular case, this means we need to move steps 1, 2, 3 into a single conversion callback function and pass it to `api_request_helper_.Request` in `swap_service.cc` - can you please confirm this?

In particular, note that step 2 merely exists in order to convert a deeply nested field using the rust lib, which looks like the following:

```json
{
  "outerArray": [
    {
      "innerArray": [
        {
          "integerFieldToConvert": 1234567890
        }
      ]
    }
  ]
```

Reply:
Yes, should be able to write your own customized conversion callback to do whatever is needed.

Reply:
Fixed here 04961e1a9d13caf1d43e13409f1c38ffa50eb5d7